presto-hive/src/main/java/com/facebook/presto/hive/HiveRecordSink.java

```
@@ -92,29 +103,170 @@
+
+        basePath = target;
+        fieldCount = columnNames.size();
+        dataFieldsCount = dataColumnNames.size();
 
-        sampleWeightField = handle.getColumnNames().indexOf(SAMPLE_WEIGHT_COLUMN_NAME);
-        columnTypes = ImmutableList.copyOf(handle.getColumnTypes());
-        connectorSession = handle.getConnectorSession();
-        hasDateTimeTypes = handle.getColumnTypes().stream().map(this::containsDateTime).collect(toList());
+        this.conf = conf;
+        this.isPartitioned = isPartitioned;
 
-        Iterable<String> hiveTypeNames = transform(transform(handle.getColumnTypes(), HiveType::toHiveType), HiveType::getHiveTypeName);
+        sampleWeightField = columnNames.indexOf(SAMPLE_WEIGHT_COLUMN_NAME);
+        this.columnTypes = columnTypes;
+        this.connectorSession = connectorSession;
+        hasDateTimeTypes = columnTypes.stream().map(this::containsDateTime).collect(toList());
 
-        Properties properties = new Properties();
-        properties.setProperty(META_TABLE_COLUMNS, Joiner.on(',').join(handle.getColumnNames()));
+        Iterable<String> hiveTypeNames = transform(transform(dataColumnTypes, HiveType::toHiveType), HiveType::getHiveTypeName);
+
+        properties = new Properties();
+        properties.setProperty(META_TABLE_COLUMNS, Joiner.on(',').join(dataColumnNames));
         properties.setProperty(META_TABLE_COLUMN_TYPES, Joiner.on(':').join(hiveTypeNames));
 
-        serializer = initializeSerializer(conf, properties, handle.getHiveStorageFormat().getSerDe());
-        recordWriter = createRecordWriter(target, conf, properties, handle.getHiveStorageFormat().getOutputFormat());
+        if (serdeParameters != null) {
+            for (String key : serdeParameters.keySet()) {
+                properties.setProperty(key, serdeParameters.get(key));
+            }
+        }
+
+        try {
+            serializer = (Serializer) lookupDeserializer(serdeLib);
```

Actually it is possible to specify different SerDe's for different partitions in Hive. Probably we should take it into account. Or at least we should throw an exception at the begging like "insert into partitioned tables with different serde settings per partition is not supported yet"